### DISCUSSION TITLE
发布流程建议：关于发版后修改安装包

### DISCUSSION THREAD

**Discussion by stormyyd** [hiroi-sora/Umi-OCR#554] [General]
作者你好！

我今天（2024/06/09）试图通过`scoop`更新`Umi-OCR`到`V2.1.2`的时候出现了报错，错误内容是因Hash校验不一致，导致安装包下载后校验失败，`scoop`进而拒绝执行下一步操作。

通过在[Scoop Extras Bucket 仓库](https://github.com/ScoopInstaller/Extras)进行了简单的搜索，发现该问题在当前版本已经出现过了两次，分别为：

- ScoopInstaller/Extras#13404
- ScoopInstaller/Extras#13410

推测是发布后又修改了两次安装包，导致Hash出现了两次异常。

GitHub这边并没有接口可以通过查询Release变更记录来直接验证这一推测，但是通过对比[V2.1.2的Release Assets](https://github.com/hiroi-sora/Umi-OCR/releases/tag/v2.1.2)信息可以发现：安装包的修改日期与源码自动打包日期、Release发布日期均存在差异。

![V2.1.2 Release](https://github.com/hiroi-sora/Umi-OCR/assets/4938213/d67ef98f-f989-41c4-b6ff-6fd35673848b)

![V2.1.2 Release Assets](https://github.com/hiroi-sora/Umi-OCR/assets/4938213/8a6d6dfd-a635-45e7-bef5-8433d3701f35)

个人建议未来发版时，如果有这种发布后需要修改安装包的需求，不要通过直接修改已发布的安装包的形式来完成，而是通过再新发一个版本的方式，已发布的版本不应该再进行修改。如果有特殊需求，比如上一个发布的版本存在重大BUG，需要避免用户误安装，也应该通过标记上一版本已废弃、删除上一个Release或清空上一个Release的Assets的方式来实现。总之就是不要在发布后修改已发布的**任何**文件。

**Comment by hiroi-sora** [comment]
非常感谢提醒！

确实，我为了修复Bug而紧急更换了发行包。我以后会避免这种行为，使用更安全的更新方式，并在发布前进行更详细的检查。

我已向scoop提交了 [PR #13426](https://github.com/ScoopInstaller/Extras/pull/13426) 来修复 Paddle 版本 Hash 不一致的问题。
